Tests/RegexTests/ParseTests.swift

```diff
@@ -600,6 +616,19 @@ extension RegexTests {
     parseTest(#"[\N{DOLLAR SIGN}-\N{APOSTROPHE}]"#, charClass(
       range_m(.namedCharacter("DOLLAR SIGN"), .namedCharacter("APOSTROPHE"))))
 
+    parseTest(
+      #"[\u{AA}-\u{BB}]"#,
+      charClass(range_m(scalar_a("\u{AA}"), scalar_a("\u{BB}")))
+    )
+    parseTest(
+      #"[\u{AA BB}-\u{CC}]"#,
+      charClass(range_m(scalarSeq_a("\u{AA}", "\u{BB}"), scalar_a("\u{CC}")))
+    )
+    parseTest(
+      #"[\u{CC}-\u{AA BB}]"#,
+      charClass(range_m(scalar_a("\u{CC}"), scalarSeq_a("\u{AA}", "\u{BB}")))
```

Comment:
What does a scalar sequence in a range entail?

Reply:
CC @natecook1000 on whether we want these inside custom ccs and whether this would do the whole NFD thing

Reply:
Let's error/reject

Reply:
Yeah, this was my frustration from the last time we discussed these. It would be great if this syntax could be used for writing a multi-scalar character in a custom CC, but that's not how it's defined at all. I agree that we should error.

Reply:
> Yeah, this was my frustration from the last time we discussed these. It would be great if this syntax could be used for writing a multi-scalar character in a custom CC, but that's not how it's defined at all. I agree that we should error.

Can we can define it that way? That makes _way_ more sense to me. But would it be a sequence again in scalar-semantics mode? Would we error out if it's more than one grapheme cluster normally?

Reply:
I think we should reject altogether in CCs for now, so that we can discuss and decide on whether there's a reasonable behavior that works for different modes.

Reply:
Updated to reject as unsupported in a custom character class for now